### PR TITLE
Allow larger tuple to vec conversions

### PR DESCRIPTION
**Pull request by sisuresh**
Required for passing args to auth module

### CODE REVIEW

**Comment by sisuresh** [comment]
@jayz22 should `impl_into_vec_for_tuple` be in events.rs? Or would it make more sense it vec.rs?

**Comment by jayz22** [comment]
@sisuresh It makes sense to move them in `vec.rs` or a new `tuple.rs`.

File: soroban-sdk/src/vec.rs [hunk 1]
```diff
@@ -40,6 +40,26 @@ macro_rules! vec {
...
+        }
+    };
+}
+impl_into_vec_for_tuple! { T0 0 }
+impl_into_vec_for_tuple! { T0 0 T1 1 }
+impl_into_vec_for_tuple! { T0 0 T1 1 T2 2 }
+impl_into_vec_for_tuple! { T0 0 T1 1 T2 2 T3 3 }
+impl_into_vec_for_tuple! { T0 0 T1 1 T2 2 T3 3 T4 4 }
+impl_into_vec_for_tuple! { T0 0 T1 1 T2 2 T3 3 T4 4 T5 5}
+impl_into_vec_for_tuple! { T0 0 T1 1 T2 2 T3 3 T4 4 T5 5 T6 6}
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply soroban-sdk/src/vec.rs]: done

File: soroban-sdk/src/vec.rs [hunk 2]
```diff
@@ -40,6 +40,32 @@ macro_rules! vec {
...
+macro_rules! impl_into_vec_for_tuple {
+    ( $($typ:ident $idx:tt)* ) => {
+        impl<$($typ),*> IntoVal<Env, Vec<RawVal>> for ($($typ,)*)
+        where
+            $($typ: IntoVal<Env, RawVal>),*
+        {
+            fn into_val(self, env: &Env) -> Vec<RawVal> {
+                vec![&env, $(self.$idx.into_val(env), )*]
+            }
+        }
```
Comment (leighmcculloch) [inline soroban-sdk/src/vec.rs:52]: Can you add conversions from &TUPLE?
```suggestion
        }
        impl<E: Env, $($typ),*> IntoVal<E, RawVal> for &($($typ,)*)
         where
             $(for<'a> &'a $typ: IntoVal<E, RawVal>),*
         {
             fn into_val(self, env: &E) -> RawVal {
                 let env = env.clone();
                 let vec = env.vec_new($count.into());
                 $(let vec = env.vec_push(vec, (&self.$idx).into_val(&env));)*
                 vec.to_raw()
             }
         }
```

Comment (sisuresh) [reply soroban-sdk/src/vec.rs:52]: Hmm the suggestion doesn't compile.

Comment (leighmcculloch) [reply soroban-sdk/src/vec.rs:52]: Let's revert it, and I'll follow up with it.

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**